src/libcore/ops.rs

```diff
@@ -1520,6 +1523,67 @@ impl<Idx: fmt::Debug> fmt::Debug for RangeTo<Idx> {
     }
 }
 
+/// An inclusive range which is bounded at both ends.
+#[derive(Copy, Clone, PartialEq, Eq)]
+#[unstable(feature = "inclusive_range", reason = "recently added, follows RFC", issue = "28237")]
+pub enum RangeInclusive<Idx> {
+    /// Empty range (iteration has finished)
+    Empty {
+        /// The point at which iteration finished
+        at: Idx
+    },
+    /// Non-empty range (iteration will yield value(s))
+    NonEmpty { // FIXME bikeshed variant name
+        /// The lower bound of the range (inclusive).
+        start: Idx,
+        /// The upper bound of the range (inclusive).
+        end: Idx,
+    },
+}
+
+#[unstable(feature = "inclusive_range", reason = "recently added, follows RFC", issue = "28237")]
+impl<Idx: fmt::Debug> fmt::Debug for RangeInclusive<Idx> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        use self::RangeInclusive::*;
+
+        match *self {
+            Empty { ref at } => write!(fmt, "[empty range @ {:?}]", at),
+            NonEmpty { ref start, ref end } => write!(fmt, "{:?}...{:?}", start, end),
+        }
+    }
+}
+
+#[unstable(feature = "inclusive_range", reason = "recently added, follows RFC", issue = "28237")]
+impl<Idx: PartialOrd + One + Sub<Output=Idx>> From<Range<Idx>> for RangeInclusive<Idx> {
+    fn from(range: Range<Idx>) -> RangeInclusive<Idx> {
+        use self::RangeInclusive::*;
+
+        if range.start < range.end {
+            NonEmpty { start: range.start, end: range.end - Idx::one() }
+        } else {
+            Empty { at: range.start } // FIXME range.start or range.end?
+        }
+    }
+}
+
+/// An inclusive range which is only bounded above.
+#[derive(Copy, Clone, PartialEq, Eq)]
+#[unstable(feature = "inclusive_range", reason = "recently added, follows RFC", issue = "28237")]
+pub struct RangeToInclusive<Idx> {
```

Comment:
Agreed, should be pretty much the same.

On Sat, Jan 16, 2016 at 3:41 PM, Steven Allen <notifications@github.com>
wrote:

> In src/libcore/ops.rs
> <https://github.com/rust-lang/rust/pull/30884#discussion_r49936902>:
>
> > +impl<Idx: PartialOrd + One + Sub<Output=Idx>> From<Range<Idx>> for RangeInclusive<Idx> {
> > +    fn from(range: Range<Idx>) -> RangeInclusive<Idx> {
> > +        use self::RangeInclusive::*;
> > +
> > +        if range.start < range.end {
> > +            NonEmpty { start: range.start, end: range.end - Idx::one() }
> > +        } else {
> > +            Empty { at: range.start } // FIXME range.start or range.end?
> > +        }
> > +    }
> > +}
> > +
> > +/// An inclusive range which is only bounded above.
> > +#[derive(Copy, Clone, PartialEq, Eq)]
> > +#[unstable(feature = "inclusive_range", reason = "recently added, follows RFC", issue = "28237")]
> > +pub struct RangeToInclusive<Idx> {
>
> It should be possible to impl From<RangeTo> for this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/30884/files#r49936902>.
>